modules/gapi/include/opencv2/gapi/core.hpp

```diff
@@ -508,6 +509,77 @@ namespace core {
             return in.withType(in.depth, in.chan).withSize(dsize);
         }
     };
+
+    G_TYPED_KERNEL(
+        GKMeansND,
+        <std::tuple<GOpaque<double>,GMat,GMat>(GMat,int,GMat,TermCriteria,int,KmeansFlags)>,
+        "org.opencv.core.kmeansND") {
+
+        static std::tuple<GOpaqueDesc,GMatDesc,GMatDesc>
+        outMeta(const GMatDesc& in, int K, const GMatDesc& bestLabels, const TermCriteria&, int,
+                KmeansFlags flags) {
+            GAPI_Assert(in.depth == CV_32F);
+            std::vector<int> amount_n_dim = detail::checkVector(in);
+            int amount = amount_n_dim[0], dim = amount_n_dim[1];
+            if (amount == -1)   // Mat with height != 1, width != 1, channels != 1 given
+            {                   // which means that kmeans will consider the following:
+                amount = in.size.height;
+                dim    = in.size.width * in.chan;
+            }
+                     // kmeans sets these labels' sizes when no bestLabels given:
+            GMatDesc out_labels(CV_32S, 1, Size{1, amount}),
+                     // kmeans always sets these centers' sizes:
+                     centers   (CV_32F, 1, Size{dim, K});
```

Comment:
Please use 2 variable's definitions instead of one (for all non-trivial variables).

This is necessary for:
- 2 dedicated lines rovided by compiler
- 2 lines in debugger
- 2 lines in valgrind reports in case of memory issues.

(+ fix indentation of comments)

```
// kmeans sets these labels' sizes when no bestLabels given:
GMatDesc out_labels(CV_32S, 1, Size{1, amount});
// kmeans always sets these centers' sizes:
GMatDesc centers   (CV_32F, 1, Size{dim, K});
```

Reply:
OK, thanks. Fixed